draft-ietf-tls-esni.md

```diff
@@ -842,6 +857,28 @@ example, if a client-facing server services many backend origin hosts, only one
 of which supports some cipher suite, it may be possible to identify that host
 based on the contents of unencrypted handshake messages.
 
+Beyond these primary security and privacy goals, ECH also aims to hide, to some
+extent, (a) whether or not a specific server supports ECH and (b) whether or
+not a specific connection uses or negotiates ECH. In particular:
+
+1. ECH support is secret from passive attackers who do not know the ECH
+configuration, but not active attackers.
+1. ECH usage is secret from passive attackers who do not know the ECH
+configuration, but not active attackers. More specifically:
+
+- Passive attackers with a known ECH configuration can distinguish between a
+connection that negotiates ECH with that configuration and one which does not,
+i.e., because the latter used a GREASE "encrypted_client_hello" extension or a
+different ECH configuration.
+- Passive attackers without the ECH configuration cannot distinguish between a
+connection that negotiates ECH and one which uses a GREASE
+"encrypted_client_hello" extension.
+- Active attackers can distinguish between a connection that negotiates ECH and
+one which uses a GREASE "encrypted_client_hello" extension.
+
+As a result, ECH is not suitable for bypassing active adversaries such as
```

Comment:
I think we should elaborate on this a bit more. Readers might take issue with the claim that ECH is not suitable for censorship circumvention. In particular, there may be deployment scenarios in which keeping the ECH configuration a secret is feasible, and "don't stick out" security against active adversaries might be possible in that setting.

Reply:
We can, but I don't think we need to do that here.

Reply:
At a minimum, I would suggest removing this line. We can have a new PR for language around censorship circumvention.

Reply:
We also have not done the analysis to know if it is possible in that setting. So, for now, I'd prefer we just be up front about this.

Reply:
Ack.